src/utils/plrStakingHelper.tsx

```diff
@@ -0,0 +1,345 @@
+  const valueInWeiString = valueInWei.toString();
+
+  // approval
+  try {
+    const abi = getContractAbi(ContractNames.ERC20Token);
+    const erc20Contract = sdk.registerContract<ERC20TokenContract>('erc20Contract', abi, plrToken?.address);
+    const approvalTransactionRequest = erc20Contract?.encodeApprove?.(contractAddress, valueInWei);
+    if (!approvalTransactionRequest || !approvalTransactionRequest.to) {
+      return { errorMessage: 'Failed to build PLR DAO stake approval transaction!' };
+    }
+
+    const approvalTransaction = {
+      to: approvalTransactionRequest.to,
+      data: approvalTransactionRequest.data,
+      chainId,
+      value: 0,
+      createTimestamp,
+      status: TRANSACTION_STATUS.UNSENT,
+    };
+    transactions = [approvalTransaction];
+  } catch (e) {
+    return { errorMessage: 'Failed to build approval transaction!' };
+  }
+
+  // staking
+  try {
+    // sdk
+    if (!stakingContract) {
+      stakingContract = sdk.registerContract<IStakingContractAbi>('PStaking', plrStakingAbi, contractAddress);
+    }
+
+    const stakeTransactionRequest = await stakingContract.encodeStake?.(valueInWeiString);
+
+    if (!stakeTransactionRequest) {
+      return { errorMessage: 'Failed build staking transaction!' };
+    }
+
+    const stakingTransaction = {
+      to: stakeTransactionRequest.to,
+      data: stakeTransactionRequest.data,
+      value: stakeTransactionRequest.value || 0,
+      chainId,
+      createTimestamp,
+      status: TRANSACTION_STATUS.UNSENT,
+    };
+
+    transactions = [...transactions, stakingTransaction];
+  } catch (e) {
+    console.log('staking error', e);
+
+    return { errorMessage: 'Failed build transfer transaction!' };
+  }
+
+  return { transactions };
+};
+
+export const estimateTransactions = async (transactions: EthereumTransaction[], feeToken: AssetOption) => {
+  etherspotService.clearTransactionsBatch(CHAIN.ETHEREUM);
+  await etherspotService.setTransactionsBatch(CHAIN.ETHEREUM, transactions);
+
+  try {
+    const estimation = await etherspotService.estimateTransactionsBatch(CHAIN.ETHEREUM, feeToken?.address);
+    return { estimation };
+  } catch (e) {
+    return { errorMessage: 'Could not estimate transaction fees.' };
+  }
+};
+
+export const sendTransactions = async (chain: string, transactionPayload: TransactionPayload) => {
+  const accountAddress = etherspotService.getAccountAddress(chain);
+
+  if (!accountAddress || !transactionPayload) return null;
+
+  try {
+    const tx = await etherspotService.sendTransaction(transactionPayload, accountAddress, chain, false);
+    return tx;
+  } catch (e) {
+    console.log('sendTransactionsError', e);
+  }
+
+  return null;
+};
+
+export const getBalanceForAddress = async (chain: string, address: string) => {
+  const sdk: Sdk = etherspotService.getSdkForChain(chain);
+
+  if (!sdk) return null;
+
+  try {
+    const { items } = await sdk.getAccountBalances({
```

Comment:
Please be aware that this endpoint regularly fails - please handle a retry if endpoint fails. If you log in to Sentry you'll see the errors

Reply:
This should be fine as this is only used in my polling call, which is called every 5 seconds when looking for a change in balance. If it fails, it'll just be called again 5 seconds later

Reply:
Ahh got it - cheers!